OrbitCore/TracepointEventBuffer.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CORE_TRACEPOINT_EVENT_BUFFER_H_
+#define ORBIT_CORE_TRACEPOINT_EVENT_BUFFER_H_
+
+#include <limits>
+#include <set>
+
+#include "SamplingProfiler.h"
+#include "capture_data.pb.h"
+
+class TracepointEventBuffer {
+ public:
+  TracepointEventBuffer() : max_time_(0), min_time_(std::numeric_limits<uint64_t>::max()) {}
+
+  std::vector<orbit_client_protos::TracepointEventInfo> GetTracepointEvents(
+      uint64_t time_begin, uint64_t time_end,
+      int32_t thread_id = SamplingProfiler::kAllThreadsFakeTid);
+
+  void AddTracepointEvent(uint64_t time, uint64_t tracepoint_hash, int32_t thread_id);
+
+  std::map<int32_t, std::map<uint64_t, orbit_client_protos::TracepointEventInfo> >
+  tracepoint_events();
+
+  Mutex& mutex();
+
+  uint64_t max_time() const;
+  uint64_t min_time() const;
+
+  bool HasEvent() const;
```

Comment:
Forgot in my last review:
You should also use that method in the TimeGraph to check for the time span.